src/scripting/game_lua_kernel.cpp

```diff
@@ -1424,6 +1424,134 @@ static int impl_end_level_data_collect(lua_State* L)
 	return 0;
 }
 
+static int impl_mp_settings_get(lua_State* L)
+{
+	void* p = lua_touserdata(L, lua_upvalueindex(1));
+	const mp_game_settings& settings = static_cast<play_controller*>(p)->get_mp_settings();
+	if(lua_type(L, 2) == LUA_TNUMBER) {
+		// Simulates a WML table with one [options] child and a variable number of [addon] children
+		// TODO: Deprecate this -> mp_settings.options and mp_settings.addons
+		size_t i = luaL_checkinteger(L, 2);
+		if(i == 1) {
+			lua_createtable(L, 2, 0);
+			lua_pushstring(L, "options");
+			lua_seti(L, -2, 1);
+			luaW_pushconfig(L, settings.options);
+			lua_seti(L, -2, 2);
+			return 1;
+		} else if(i >= 2) {
+			i -= 2;
+			if(i < settings.addons.size()) {
+				auto iter = settings.addons.begin();
+				std::advance(iter, i);
+				config cfg;
+				iter->second.write(cfg);
+				cfg["id"] = iter->first;
+
+				lua_createtable(L, 2, 0);
+				lua_pushstring(L, "addon");
+				lua_seti(L, -2, 1);
+				luaW_pushconfig(L, cfg);
+				lua_seti(L, -2, 2);
+				return 1;
+			}
+		}
+	} else {
+		char const *m = luaL_checkstring(L, 2);
+		return_string_attrib("scenario", settings.name); // TODO: Deprecate this -> mp_settings.game_name
+		return_string_attrib("game_name", settings.name);
+		return_string_attrib("hash", settings.hash);
+		return_string_attrib("mp_era_name", settings.mp_era_name); // TODO: Deprecate this -> scenario.era.name
+		return_string_attrib("mp_scenario", settings.mp_scenario); // TODO: Deprecate this -> scenario.id
+		return_string_attrib("mp_scenario_name", settings.mp_scenario_name); // TODO: Deprecate this -> scenario.name (which doesn't exist yet)
+		return_string_attrib("mp_campaign", settings.mp_campaign); // TODO: Deprecate this -> scenario.campaign.id
+		return_string_attrib("side_users", utils::join_map(settings.side_users)); // TODO: Deprecate this -> mp_settings.side_players
```

Comment:
Most of these todos look like more "moving stuff because you don't like the current name".

Reply:
I don't want these deprecated.  They're supposed to still exist and people might be using them, so just leave them be.